lib/miq_automation_engine/engine/miq_ae_engine/drb_remote_invoker.rb

```diff
@@ -120,7 +120,7 @@ class AutomateMethodException < StandardError
   # undefine Object#display which would be called over service#display
   DRbObject.send(:undef_method, :display)
 
-  DRb.start_service
+  DRb.start_service("druby://localhost:0")
```

Comment:
@jrafanie Can we use a net stat like command to see how many sockets are opened and by which processes. Since the client is running as a separate process we should be able to get the port it is using.
The other thing is if we are using pipes to communicate from engine (server) to client, shouldn't we use pipes when communicating from client to engine (server). Not sure if this is an an Out Of Band messaging scheme from client to server. How would the engine (server side) know which client to contact. When we launch a method we set the MIQ_URI to tell the client where the server is listening, but how does the client tell the engine where his server is listening.

Reply:
yeah, I was wondering if we should just use unix sockets on the "client" side of the drb communication too.  I'll change it to do that.  

My understanding is that both our automate server and client both start drb servers for communication but only the client requests the server's drb object but both need drb servers started.  I tried without `start_service` and it fails.

Reply:
On my machine it creates a TCP socket:

```
irb(main):001:0> r = DRb.start_service
=> #<DRb::DRbServer:0x00007f947fcc0a90 @config={:idconv=>#<DRb::DRbIdConv:0x00007f94801940d0>, :verbose=>false, :tcp_acl=>nil, :load_limit=>26214400, :argc_limit=>256, :safe_level=>0}, @protocol=#<DRb::DRbTCPSocket:0x00007f947fcc04a0 @uri="druby://92.242.140.21:54402", @socket=#<TCPServer:fd 37, AF_INET, 0.0.0.0, 54402>, @config={:tcp_original_host=>"", :idconv=>#<DRb::DRbIdConv:0x00007f94801940d0>, :verbose=>false, :tcp_acl=>nil, :load_limit=>26214400, :argc_limit=>256, :safe_level=>0, :tcp_port=>54402}, @acl=nil, @msg=#<DRb::DRbMessage:0x00007f947fcc0478 @load_limit=26214400, @argc_limit=256>, @shutdown_pipe_r=#<IO:fd 38>, @shutdown_pipe_w=#<IO:fd 39>>, @uri="druby://92.242.140.21:54402", @exported_uri=["druby://92.242.140.21:54402"], @front=nil, @idconv=#<DRb::DRbIdConv:0x00007f94801940d0>, @safe_level=0, @grp=#<ThreadGroup:0x00007f947fcc0388>, @thread=#<Thread:0x00007f947fcc0338@/Users/joerafaniello/.rubies/ruby-2.6.5/lib/ruby/2.6.0/drb/drb.rb:1504 sleep>>
irb(main):002:0> r.uri
=> "druby://92.242.140.21:54402"
irb(main):003:0> Process.pid
=> 69469

```

```
> lsof -i -P |grep 69469
ruby      69469 joerafaniello   32u  IPv4 0x9455ca3322cc180b      0t0  TCP localhost:54395->localhost:11211 (ESTABLISHED)
ruby      69469 joerafaniello   37u  IPv4 0x9455ca3322cc7013      0t0  TCP *:54402 (LISTEN)
```